packages/flutter/test/material/expansion_panel_test.dart

```diff
@@ -1247,4 +1247,432 @@ void main() {
+                      return Checkbox(
+                        value: isExpanded,
+                        onChanged: (bool _) {
+                          isTriggered = true;
+                          handlePressed();
+                        },
+                      );
+                    },
+                    body: Text(_isExpanded[0] ? 'B' : 'A'),
+                  ),
+                  ExpansionPanelRadio(
+                    value: 1,
+                    canTapOnHeader: true,
+                    headerBuilder: (BuildContext context, bool isExpanded) {
+                      return const Text('Header', key: secondPanelKey);
+                    },
+                    expandIconBuilder: (
+                      BuildContext context,
+                      bool isExpanded,
+                      VoidCallback handlePressed,
+                      Duration animationDuration,
+                    ) {
+                      return Checkbox(
+                        value: isExpanded,
+                        onChanged: (bool _) {
+                          isTriggered = true;
+                          handlePressed();
+                        },
+                      );
+                    },
+                    body: Text(_isExpanded[1] ? 'D' : 'C'),
+                  ),
+                ],
+              );
+            },
+          ),
+        ),
+      ),
+    );
+
+    expect(find.byType(Checkbox), findsNWidgets(2));
+    expect(find.byType(ExpandIcon), findsNothing);
+
+    expect(find.text('A'), findsNothing);
+    expect(find.text('B'), findsOneWidget);
+    expect(find.text('C'), findsOneWidget);
+    expect(find.text('D'), findsNothing);
+    expect(
+      tester.widget<Checkbox>(find.byType(Checkbox).at(0)).value,
+      isTrue,
+    );
+    expect(
+      tester.widget<Checkbox>(find.byType(Checkbox).at(1)).value,
+      isFalse,
+    );
+
+    // should not trigger onChanged, but should expand
+    await tester.tap(find.byType(Checkbox).at(1));
+    await tester.pumpAndSettle();
+
+    expect(find.text('A'), findsOneWidget);
+    expect(find.text('B'), findsNothing);
+    expect(find.text('C'), findsNothing);
+    expect(find.text('D'), findsOneWidget);
+    expect(isTriggered, isFalse);
+    expect(
+      tester.widget<Checkbox>(find.byType(Checkbox).at(0)).value,
+      isFalse,
+    );
+    expect(
+      tester.widget<Checkbox>(find.byType(Checkbox).at(1)).value,
+      isTrue,
+    );
+
+    // should expand first panel by tapping on header
+    await tester.tap(find.byKey(firstPanelKey));
+    await tester.pumpAndSettle();
+
+    expect(find.text('A'), findsNothing);
+    expect(find.text('B'), findsOneWidget);
+    expect(find.text('C'), findsOneWidget);
+    expect(find.text('D'), findsNothing);
+    expect(
+      tester.widget<Checkbox>(find.byType(Checkbox).at(0)).value,
+      isTrue,
+    );
+    expect(
+      tester.widget<Checkbox>(find.byType(Checkbox).at(1)).value,
+      isFalse,
+    );
+  });
```

Comment:
Can you add a test that test the semantics flag and label ?

Reply:
Done